source/contracts/libraries/token/VariableSupplyToken.sol

```diff
@@ -30,6 +30,18 @@ contract VariableSupplyToken is StandardToken {
     function burn(address _target, uint256 _amount) internal returns (bool) {
         balances[_target] = balances[_target].sub(_amount);
         supply = supply.sub(_amount);
+        onBurn(_target, _amount);
+        return true;
+    }
+
+    // Subclasses of this token may want to send additional logs through the centralized Augur log emitter contract
+    function onMint(address _target, uint256 _amount) internal returns (bool) {
+        Mint(_target, _amount);
```

Comment:
Similar to how `Transfer` event should be fired independent of overrides, I think the `Mint`/`Burn` functions should be fired independent of overrides.  That is, we should always fire a `Mint`/`Burn` event from the token contract, as well as doing whatever else the derived contract wants to do on mint/burn (such as firing another event from Augur.sol.

The argument here is weaker, since Mint/Burn are not part of the ERC20 spec, but I think there is value in them being readily available to things like chain explorers, especially for REP which will most likely be a well known contract address.

Reply:
TL;DR: move `Mint` and `Burn` out of the `onMint` and `onBurn` functions and into the `mint` and `burn` functions right before the `onMint`/`onBurn` function is called.

Reply:
Should we do that for `Transfer` as well? currently that is not the pattern being used

Reply:
Yes, I believe @bthaile is working on a PR to do that as part of https://app.clubhouse.io/augur/story/4451/fixup-token-transfers

Reply:
I realize now that you may not have read that CH issue prior to me mentioning it just now.  😛